substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JDKConfigFiles.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.hosted.jdk;
+
+import com.oracle.svm.core.BuildArtifacts;
+import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
+import com.oracle.svm.core.feature.InternalFeature;
+import com.oracle.svm.core.util.VMError;
+import org.graalvm.nativeimage.ImageSingletons;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public final class JDKConfigFiles {
+
+    public static JDKConfigFiles singleton() {
+        return ImageSingletons.lookup(JDKConfigFiles.class);
+    }
+
+    final Set<String> configFiles = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    JDKConfigFiles() {
+    }
+
+    public void register(String configFile) {
+        if (configFile.startsWith("/")) {
+            configFiles.add(configFile.substring(1));
+        } else {
+            configFiles.add(configFile);
+        }
```

Comment:
How about this?
```suggestion
            configFiles.add(configFile.startsWith("/") ? configFile.substring(1) : configFile);
```
Also, shouldn't we use `File.separator` instead of `"/"` here?

Reply:
The thing that's validated here is user input, so it all depends on how `JDKConfigFiles.singleton().register("")` is called.
In theory, removing the leading slash isn't required either if nobody passes one in.. So depends on how much input sanitisation you want to do.

Reply:
Right, how about changing this to something like `register(String... configPathParts)`, which can then be used like `register("lib", "bar", "my.file");`. Parts are not allowed to be `..` or to start with `/` or `\`. `register()` then joins the parts using the correct `File.separator` and validates that the joined path exists in `java.home`. If any of this is not the case, fail with an error.

Reply:
You could even just use `Paths.get()` in `register()`.

Reply:
A vararg does make more sense indeed. I've modified it and changed the validation accordingly.